script.js

```diff
@@ -1,4 +1,142 @@
+After both players have rolled and chosen dice order, the player with the higher combined number wins.*/
+
+//set global variables to game mode
+var GAMEMODE_WAITING_FOR_NAME1 = "waiting for player 1 to key in name";
+var GAMEMODE_WAITING_FOR_NAME2 = "waiting for player 2 to key in name";
+var GAMEMODE_PLAYER1 = "Player 1 mode";
+var GAMEMODE_PLAYER2 = "Player 2 mode";
+var GAMEMODE_T0_CHOOSE_ORDER = "For player to choose order of dice";
+var TO_CHOOSE_WINNER = "Compare player 1 and player dice results";
+var GAMEMODE_T0_CHOOSE_PLAYER = "either player 1 or 2";
+
+var GAME_STATE = GAMEMODE_WAITING_FOR_NAME1;
+var GAMEPLAYER = GAMEMODE_PLAYER1;
+var player1CombinedNumber = 0;
+var player2CombinedNumber = 0;
+
+//-----------------------------------------------------------------------------------
+//diceroll1 and diceroll2 corresponds to player 1 dice roll results
+var diceRoll1 = 0;
+var diceRoll2 = 0;
+
+const rollDice = () => (randomNumber = Math.floor(Math.random() * 6 + 1));
+
+//output to show the results of 2 dice roll
+const showsDiceRollsResult = () => {
+  let diceRollsresult = [];
+  for (let i = 0; i < 2; i++) {
+    var singleDiceRollresult = rollDice();
+    diceRollsresult.push(singleDiceRollresult);
+  }
+
+  //diceRollsresult --> store an array with 2 diceroll result
+  diceRoll1 = diceRollsresult[0];
+  diceRoll2 = diceRollsresult[1];
+
+  return `This is the result of your dice rolls <br> ------------------------------------ <br> DiceRoll 1: ${diceRoll1} <br> DiceRoll 2: ${diceRoll2} `;
+};
+
+//function to show game player 1 or game player 2 choice
+const showGamePlayerChoice = (GAMEPLAYER, userChoiceOrder) => {
+  GAME_STATE = GAMEMODE_WAITING_FOR_NAME2;
+
+  if (GAMEPLAYER == GAMEMODE_PLAYER1) {
+    if (userChoiceOrder == 1) {
+      player1CombinedNumber = Number(`${diceRoll1}${diceRoll2}`);
+      return `Your number is ${player1CombinedNumber}. It's now player 2's turn. Player 2 please enter your name. `;
+    }
+    if (userChoiceOrder == 2) {
+      player1CombinedNumber = Number(`${diceRoll2}${diceRoll1}`);
+      return `Your number is ${player1CombinedNumber}. It's now player 2's turn. Player 2 please enter your name.`;
+    }
+  }
+
+  if (GAMEPLAYER == GAMEMODE_PLAYER2) {
+    GAME_STATE = TO_CHOOSE_WINNER;
+    if (userChoiceOrder == 1) {
+      player2CombinedNumber = Number(`${diceRoll1}${diceRoll2}`);
+      return `Player 2: Your number is ${player2CombinedNumber} Now let's see who wins!.Please click submit to see the result`;
+    }
+    if (userChoiceOrder == 2) {
+      player2CombinedNumber = Number(`${diceRoll2}${diceRoll1}`);
+      return `Player 2: Your number is ${player2CombinedNumber}.Now let's see who wins! Please click submit to see the result`;
+    }
+  }
+};
+
+const toCompareNumber = (player1number, player2number) => {
+  return player1number == player2number
+    ? "It is a draw"
```

Comment:
Good use of Ternary Operators.. Personally i do not prefer using Ternary Operators due to the fact that it is hard to understand at a first glance. But i really do appreciate you going above and beyond the course.